wordpress-mobile/WordPress-Android
# Jetpack Section: Remove Jetpack Feature Flags

Fixes #13267

This PR is responsible for removing all `Jetpack Section` related feature flags from the codebase. After that, the whole `Jetpack Section` grandparent related issue (see fixes) can be closed since those are the last items on its todo list.

To test:
- Launch the app and go to `My Site` tab.
- Test the `Activity Log` and `Backup` screens and more specifically make sure the `Activity Log Filters` are available and functional.
- Test the `Backup` screen individually and more specifically make sure the screen is available and functional.
- Test the `Scan` screen and more specifically make sure the screen is available and functional.
- Test the `Back Download` screen and flows from within the `Activity Log` and `Backup` screens.
- Test the `Restore` screen and flows from within the `Activity Log` and `Backup` screens.

Merge instructions:
- Wait for confirmation from @leandroalonso before merging.
- Wait for a 👍 from @malinajirka as well.

## Regression Notes
1. Potential unintended areas of impact

N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

N/A

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:
